File: backend/src/main/java/br/com/calciolari/datahub/imports/infrastructure/storage/RawFileDescriptor.java
package br.com.calciolari.datahub.imports.infrastructure.storage;

import java.util.Objects;

/**
 * Descriptor for an immutable raw artifact. Storage keys are server-generated;
 * never derived from {@code originalFilename}.
 */
public record RawFileDescriptor(
		String sha256,
		long byteSize,
		String detectedType
) {
	public RawFileDescriptor {
		Objects.requireNonNull(sha256, "sha256");
		if (sha256.length() != 64) {
			throw new IllegalArgumentException("sha256 must be 64 hex chars");
		}
		if (byteSize < 0) {
			throw new IllegalArgumentException("byteSize must be >= 0");
		}
	}
}

File: backend/src/Calciolari.DataHub/Imports/Infrastructure/Storage/RawFileDescriptor.cs
namespace Calciolari.DataHub.Imports.Infrastructure.Storage;

/// <summary>
/// Descriptor for an immutable raw artifact. Storage keys are server-generated;
/// never derived from originalFilename.
/// </summary>
public sealed record RawFileDescriptor(string Sha256, long ByteSize, string? DetectedType)
{
    public static RawFileDescriptor Create(string sha256, long byteSize, string? detectedType)
    {
        ArgumentNullException.ThrowIfNull(sha256);
        if (sha256.Length != 64)
        {
            throw new ArgumentException("sha256 must be 64 hex chars", nameof(sha256));
        }

        if (byteSize < 0)
        {
            throw new ArgumentOutOfRangeException(nameof(byteSize), "byteSize must be >= 0");
        }

        return new RawFileDescriptor(sha256, byteSize, detectedType);
    }
}
